qiskit/primitives/__init__.py

```diff
@@ -17,59 +17,181 @@
+
+2. The V2 sampler returns samples of classical outcomes, preserving the shot order in which they 
+   were measured. This is in contrast to the V1 sampler that outputs quasi-probability distributions
+   which are instead an *estimate of the distribution* over classical outcomes. Moreover, the V2
+   sampler result objects organize data in terms of their input circuits' classical register 
+   names, which provides natural compatibility with dynamic circuits.
+
+   .. code-block:: python
+
+      circuit = QuantumCircuit(QuantumRegister(2, "qreg"), ClassicalRegister(2, "alpha"))
+      circuit.h(0)
+      circuit.cx(0, 1)
+      circuit.measure([0, 1], [0, 1])
+
+      # V1 sampler usage
+      result = sampler_v1.run([circuit]).result()
+      quasi_dists = bitstrings.get_counts()
+
+      # V2 sampler usage
+      result = sampler_v2.run([circuit]).result()
+      # these are all the bitstrings from the alpha register
+      bitstrings = result[0].data.alpha
+      # we can use it to generate a Counts mapping, which is the most similar thing to a quasi dist
+      counts = bitstrings.get_counts()           
+
+3. The V2 primitives have brought the concept of sampling overhead, inherent to all quantum systems 
+   via their inherent probabilistic nature, out of the options and into the API itself. For the
+   sampler, this means that the ``shots`` argument is now part of the :meth:`~.BaseSamplerV2.run`
+   signature, and moreover that each pub is able to specify its own value for ``shots``, which takes
+   precedence over any value given to the method. The sampler has an analogous ``precision``
+   argument that specifies the error bars that the primitive implementation should target for
+   expectation values estimates.
+
+   This concept is not present in the API of the V1 primitives, though all implementations of the 
+   V1 primitives have related settings somewhere in their options.
+
+   .. code-block:: python
+
+      # Sample two circuits at 128 shots each.
+      sampler_v2.run([circuit1, circuit2], shots=128)
+
+      # Sample two circuits at different amounts of shots. The "None"s are necessary as placeholders
+      # for the lack of parameter values in this example.
+      sampler_v2.run([(circuit1, None, 123), (circuit2, None, 456)])
+
+      # Estimate expectation values for two pubs, both with 0.05 precision.
+      estimator_v2.run([(circuit1, obs_array1), (circuit2, obs_array_2)], precision=0.05)
 
-.. autosummary::
-   :toctree: ../stubs/
 
-   BaseEstimator
-   Estimator
-   BackendEstimator
 
-EstimatorV2
-===========
+Primitives API
+==============
+
+Primitives V2
+-------------
 
 .. autosummary::
    :toctree: ../stubs/
 
+   BaseEstimatorV2
    StatevectorEstimator
 
-Sampler
-=======
+Sampler V2
+----------
 
 .. autosummary::
    :toctree: ../stubs/
 
-   BaseSampler
-   Sampler
-   BackendSampler
    BaseSamplerV2
    StatevectorSampler
 
-Results
-=======
+Containers V2
```

Comment:
Containers is a bit jargony for me for a doc heading. Results V2 might be better
```suggestion
Results V2
```